src/Scripting/CSharp/CSharpScript.cs

```diff
@@ -195,15 +200,34 @@ protected override Compilation CreateCompilation()
 
             var references = this.GetReferencesForCompilation();
 
-            var parseOptions = this.Options.IsInteractive ? s_defaultInteractive : s_defaultScript;
-            var tree = SyntaxFactory.ParseSyntaxTree(this.Code, parseOptions, path: this.Path);
+            SyntaxTree[] trees;
+            var diagnostics = DiagnosticBag.GetInstance();
+            try
+            {
+                trees = GetSyntaxTrees(
+                    this.Code,
+                    this.Path,
+                    this.Options.IsInteractive,
+                    diagnostics,
+                    SourceFileResolver.Default,
+                    CSharpScript<T>.DefaultFileReader);
```

Comment:
Type qualifier is unnecessary. And do we need to expose DefaultFileReader or could fileReader be an optional argument where DefaultFileReader is used if null?